src/components/EditorHeader/ControlPanel.jsx

```diff
@@ -1211,6 +1211,17 @@ export default function ControlPanel({
         function: resetView,
         shortcut: "Ctrl+R",
       },
+      show_datatype: {
+        state: settings.showDataTypes ? (
+          <i className="bi bi-toggle-on" />
+        ) : (
+          <i className="bi bi-toggle-off" />
+        ),
+        function: () => {
```

Comment:
```suggestion
     function: () =>
          setSettings((prev) => ({
            ...prev,
            showDataTypes: !prev.showDataTypes,
          })),
```
Please make sure to update the state using the setSettings function. Also  the build fails here, please take a lokk https://github.com/drawdb-io/drawdb/actions/runs/13877624768/job/38843669905?pr=366